### PR TITLE
Add external sources resource

### PR DESCRIPTION
**Pull request by syncush**
@galta95 please fill this form

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| New feature     | ✔                                                                       |
| Tests added     | ✔                                                                       |

### CODE REVIEW

File: src/sources/models/converters.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  return Promise.resolve(converted);
+}
+
+export async function geoJsonConverter(geometry: unknown): Promise<Geometry> {
+  if (!isGeoJson(geometry)) {
+    throw new GeomertyParseError(`Cannot convert geometry to geoJson geometry`);
+  }
+  return Promise.resolve(geometry);
+}
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/sources/models/converters.ts:32]: geoJsonConverter is bad name selection, I will change it. I meant  geoJsonGeometryConverter (this function refer only to the geometry).

File: .gitignore [hunk 2]
```diff
@@ -107,3 +107,4 @@ dist
 # Jest
 jest_html_reporters.html
 reports
+schema.json
```
[the remark this answers is not in the document]
Comment (galta95) [reply .gitignore:112]: Done

File: README.md [hunk 3]
```diff
@@ -1,47 +1,3 @@
...
-```
-
-usage example:
-
-```typescript
-public constructor(
-    @inject(delay(() => MCLogger)) private readonly logger: MCLogger) {
-  }
-```
+Service for convert external format for geoJson format
```
[the remark this answers is not in the document]
Comment (galta95) [reply README.md]: Done

File: package.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
     "prebuild": "npm run clean",
     "build": "tsc --project tsconfig.build.json  && npm run assets:copy",
     "start": "npm run build && cd dist && node ./index.js",
-    "assets:copy": "copyfiles -f ./config/* ./dist/config && copyfiles -f ./openapi3.yaml ./dist/ && copyfiles ./package.json dist",
+    "assets:copy": "copyfiles -f ./config/* ./dist/config && copyfiles -f ./openapi3.yaml ./dist/ && copyfiles ./package.json dist && copyfiles -f ./validationJsonSchema.json ./dist/ && copyfiles -f ./schema.json ./dist/ ",
```
[the remark this answers is not in the document]
Comment (galta95) [reply package.json:20]: Done

File: src/serverBuilder.ts [hunk 5]
```diff
@@ -35,14 +35,14 @@ export class ServerBuilder {
   private buildRoutes(): void {
     this.logger.log('debug', 'registering service routes');
 
-    this.serverInstance.use('/resourceName', resourceNameRouterFactory(container));
+    this.serverInstance.use('/sources', sourcesRouterFactory(container));
     this.serverInstance.use('/', swaggerRouterFactory(container));
+    this.serverInstance.use(this.errorHandler.getErrorHandlerMiddleware());
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/serverBuilder.ts]: Done

File: src/sources/models/converters.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+export async function wktConverter(geometry: unknown): Promise<Geometry> {
+  if (!(typeof geometry === 'string')) {
+    throw new GeomertyParseError(`Cannot convert geometry to geoJson geometry`);
+  }
+  const converted = parse(geometry);
+  if (converted === null) {
+    throw new GeomertyParseError(`Cannot convert geometry to geoJson geometry`);
+  }
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/sources/models/converters.ts]: Done

File: src/sources/models/sourcesManager.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+@injectable()
+export class SourcesManager {
+  public constructor(
+    @inject(Services.LOGGER) private readonly logger: ILogger,
+    @inject(Schemas) private readonly schemas: Schemas,
+    @inject('converters') private readonly converters: Converters
+  ) {}
+
+  public async convert(body: IBodyRecord, sourceName: string): Promise<Feature> {
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/sources/models/sourcesManager.ts]: Done

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (syncush) [reply package.json:20]: you copy the assets twice

File: src/sources/controllers/sources.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    } catch (e) {
+      const error = e as HttpError;
+      if (e instanceof SchemaNotFoundError) {
+        error.statusCode = httpStatus.NOT_FOUND;
+      } else if (e instanceof GeomertyParseError) {
+        error.statusCode = httpStatus.BAD_REQUEST;
+      } else if (e instanceof GeometryNotFoundError) {
+        error.statusCode = httpStatus.UNPROCESSABLE_ENTITY;
+      }
+      this.logger.log('error', error.message, e);
```
Comment (syncush) [inline src/sources/controllers/sources.ts:37]: error handler also logs the errors, this is redundant

File: src/sources/models/converters.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  if (converted === null) {
+    throw new GeomertyParseError(`Cannot convert geometry to GeoJson geometry`);
+  }
+  return Promise.resolve(converted);
+}
+
+export async function geoJsonConverter(geometry: unknown): Promise<Geometry> {
+  if (!isGeoJson(geometry)) {
+    throw new GeomertyParseError(`Cannot convert geometry to GeoJson geometry`);
+  }
```
Comment (syncush) [inline src/sources/models/converters.ts:30]:
```suggestion
  if (!isGeoJsonGeometry(geometry)) {
    throw new GeomertyParseError(`invalid geometry, geometry is not according to GeoJSON RFC`);
  }
```

File: tests/integration/sources/helpers/testSchema.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
+[
+  {
+    "name": "animels",
```
Comment (syncush) [inline tests/integration/sources/helpers/testSchema.json:3]: typo

File: tests/unit/resourceName/models/sourcesManager.spec.ts [hunk 11]
```diff
@@ -0,0 +1,225 @@
+import { SourcesManager } from '../../../../src/sources/models/sourcesManager';
```
Comment (syncush) [inline tests/unit/resourceName/models/sourcesManager.spec.ts:1]: folder hierarchy is having resourceName

File: tests/integration/sources/sources.spec.ts [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    registerTestValues();
+    await requestSender.init();
+  });
+  afterEach(function () {
+    container.clearInstances();
+  });
+
+  describe('Happy Path', function () {
+    it('should return 200 status code and external data converted to GeoJson with wkt geometry', async function () {
+      const externalSource = 'animels';
```
Comment (syncush) [inline tests/integration/sources/sources.spec.ts:20]: typo

Comment (galta95) [reply tests/integration/sources/sources.spec.ts:20]: Done

File: tests/unit/resourceName/models/sourcesManager.spec.ts (hunk 11 above)
Comment (galta95) [reply tests/unit/resourceName/models/sourcesManager.spec.ts:1]: Done

File: tests/integration/sources/helpers/testSchema.json (hunk 10 above)
Comment (galta95) [reply tests/integration/sources/helpers/testSchema.json:3]: Done

File: src/sources/models/converters.ts (hunk 9 above)
Comment (galta95) [reply src/sources/models/converters.ts:30]: Done

File: src/sources/controllers/sources.ts (hunk 8 above)
Comment (galta95) [reply src/sources/controllers/sources.ts:37]: Done

**Pull request merged**